### ISSUE TITLE
How can I set Application Badge Number

### ISSUE THREAD

**Issue by jafsab** [arnesson/cordova-plugin-firebase#1110]
I have ionic 3 project your using cordova-plugin-firebase plugin.
I want to increase the badge number by 1 when new notification come from firebase.

I want to display the badge on the application icon on mobile home screen.
like the screenshot below.

![b1](https://user-images.githubusercontent.com/9292442/65147344-a007f800-da26-11e9-9849-d5b42503d54f.jpeg)
![b2](https://user-images.githubusercontent.com/9292442/65147358-a72f0600-da26-11e9-8fea-9f2f0fb5d4d0.jpeg)

**Comment by FrancescoPaiola** [comment]
i'm using this [plugin](https://ionicframework.com/docs/v3/native/badge/) it works, but i can't manage to update the badge while app is in background or closed ._.

**Comment by jafsab** [comment]
yes Francesco Im asking the same, I need to update it while the app in background

**Comment by FrancescoPaiola** [comment]
A possible solution (i can't check it right now) is to send the notification with `badge` attribute.
Search badge [here](https://firebase.google.com/docs/cloud-messaging/http-server-ref).

**Comment by uokivan** [comment]
> A possible solution (i can't check it right now) is to send the notification with `badge` attribute.
> Search badge [here](https://firebase.google.com/docs/cloud-messaging/http-server-ref).

but this way on iOS, it can't calculate the badge, such as : if the badge now is 1 , the best way is show "2" when the app get one more message, but it can't

**Comment by FrancescoPaiola** [comment]
You could store notifications on a custom model server side with a `read` flag that has to be True every time you open a notification. 
Than you have to increase the badge attribute server side for each new notification, decrease it every time you mark a notification read flag as False. 
Could be a workaround?

**Comment by uokivan** [comment]
> You could store notifications on a custom model server side with a `read` flag that has to be True every time you open a notification.
> Than you have to increase the badge attribute server side for each new notification, decrease it every time you mark a notification read flag as False.
> Could be a workaround?

Sure, I thing that is the way we can  do

**Comment by jafsab** [comment]
thanks for all, I will try it , this way for ios! 
what about android? how can we do it?
